sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -998,6 +998,22 @@ class Analyzer(
   }
 
   /**
+   * Replace unresolved expressions in grouping keys with resolved ones in SELECT clauses.
+   */
+  object ResolveAggAliasInGroupBy extends Rule[LogicalPlan] {
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperators {
+      case agg @ Aggregate(groups, aggs, child)
+          if conf.groupByAliases && child.resolved && aggs.forall(_.resolved) &&
```

Comment:
I'm not very confident about this condition, we may mistakenly resolve grouping expression by aggregate list while it should be resolved by child output.

one example is the star. If the aggregate list contains a star, then we will expand the star in `ResolveReferences`, without resolving grouping expressions. When we reach here, the condition will match but the grouping expression should not be resolved by aggregate list.

cc @gatorsmile